x-pack/plugins/security/server/config_deprecations.ts

```diff
@@ -13,6 +13,7 @@ export const securityConfigDeprecationProvider: ConfigDeprecationProvider = ({
   unused,
 }) => [
   rename('sessionTimeout', 'session.idleTimeout'),
+  rename('authProviders', 'authc.providers'),
```

Comment:
It's not clear to me why you're mapping between `authProviders` and `authc.providers` - is this reflected in the documentation?

Reply:
Yeah, it's [in our old docs](https://www.elastic.co/guide/en/kibana/6.8/kibana-authentication.html). The `authProviders: string[]` was the first authc-related configuration we introduced in 6.2/3, then we migrated to a [`authc.providers: string[]`](https://www.elastic.co/guide/en/kibana/7.3/kibana-authentication.html) in anticipation to support a more complex `authc.providers: Record<string, MoreComplexStructure>` in the future (that's what we have today).

With this `rename` we just continue to support this ancient `authProviders: string[]` format.

Reply:
Our deprecation handling logic will transform this:
```
xpack.security.authProviders: [basic]
```
first to this:
```
xpack.security.authc.providers: [basic]
```
and then to this:
```
xpack.security.authc.providers.basic.basic.order: 0
```